### DISCUSSION TITLE
Extended public key support

### DISCUSSION THREAD

**Discussion by armas-mk** [iotaledger/firefly#1530] [Features / Enhancements]
## Description

Generate a public key for a wallet that can be used to query nodes to show the available balance in an aggregated output.

Same as xPub key in many popular cryptocurrencies.

## Motivation

It would make it easier to integrate IOTA wallets with tracker apps (eg.: cointracker.io). I'm sure there are plenty of other use cases as well.

## Requirements

Write a list of what you want this feature to do.

1. Generate external public key or a master view public key

## Open questions (optional)

I assume it will also require some changes into the node software's API. Shall i create a FEATURE request there as well with a link to this issue?

## Are you planning to do it yourself in a pull request?

No. Sorry, I don't have the time.

**Comment by cvarley100** [comment]
Unfortunately, I don't think xPub is available for our signature scheme (cc @Wollac).

**Comment by Wollac** [comment]
Yes, xPub is defined for Bitcoin's bip-32 but not slip-10 with Ed25519 which is e.g. used in IOTA.

There are some ideas how to implement a similar scheme, but nothing is standardized or vetted enough yet.
We will of course revisit this once this has changed.

**Comment by andywolfman999** [comment]
my public key its error.. wtf?
wallet adress is not corect.. 
Firefly v1.3.0
atoi1qq9rw05efvy4u25p0g90fd7tntygch76rxnc8w8n60n0wegmnyp7uk808zs

first word.. need to be iota1.. I can not transfer iota to this adress

**Reply by t00k** [reply]
Do not use developer profiles in the Firefly Wallet if you are not a Developer. A developer Profile will connect you to the Testnet and generate addresses that do not work on IOTA Mainnet. These testnet addresses start with "atoi1" instead of the mainnet addresses "iota1".
If you have accidentally created a developer profile and did send funds to an "atoi1" address you have not lost your funds. 
To make these funds accessible you need to use the "recover a wallet" function in Firefly while creating a new profile. 
Create a "non developer" profile and use the 24 words that you have saved from your "developer profile" to restore this new profile (non developer).
Your funds will show up in this new profile.
